rotkehlchen/exchanges/coinbase.py

```diff
@@ -160,21 +161,117 @@ def __init__(
             secret=secret,
             database=database,
         )
+        self.is_legacy_api_key=self.determine_key_type(api_key)
         self.apiversion = 'v2'
         self.base_uri = 'https://api.coinbase.com'
         self.msg_aggregator = msg_aggregator
-        self.session.headers.update({'CB-ACCESS-KEY': self.api_key})
-        self.session.headers.update({'CB-VERSION': CB_VERSION})
+        self.host = 'api.coinbase.com'
+
+        
+        if self.is_legacy_api_key:
+            self.session.headers.update({'CB-ACCESS-KEY': self.api_key})
+            self.session.headers.update({'CB-VERSION': CB_VERSION})
+
+    def determine_key_type(self, api_key)-> bool:
+        """
+        Determines if the provided API key and secret are legacy or new.
+        Returns True if legacy, False if new.
+        """
+        legacy_pattern = r'^[\w]+$' 
+        new_pattern = r'^organizations/[\w-]+/apiKeys/[\w-]+$'
+
+        if re.match(legacy_pattern, api_key):
+            log.debug("Legacy Key format!")
+            return True
+        elif re.match(new_pattern, api_key):
+            log.debug("New Key format!")
+            return False
+        else:
+            raise ValueError(f'Invalid API key format: {api_key}')
```

Comment:
Our codebase never raises ValueErrors. This would end up being unhandled and crash the application.

Use `AuthenticationError` from `rotkehlchen/errors/api.py`.

Also:

1. Add it in the function's docstring it may raise this.
2. Hande it wherever it's raised.